same as 42 but then for version, tidied up a bit, and use hunt_server_cmd to find and forward remote requests to, rather than doing that work itself in the various *_version functions.

diff -r 6bd3926f94aa ircd/m_version.c
--- a/ircd/m_version.c	Sat Jan 31 18:42:09 2009 +0100
+++ b/ircd/m_version.c	Sat Jan 31 18:43:26 2009 +0100
@@ -109,7 +109,7 @@
  */
 int m_version(struct Client* cptr, struct Client* sptr, int parc, char* parv[])
 {
-  if (parc > 1 && match(parv[1], cli_name(&me)))
+  if (parc > 1)
     return send_reply(sptr, ERR_NOPRIVILEGES);
 
   send_reply(sptr, RPL_VERSION, version, debugmode, cli_name(&me),
@@ -126,29 +126,18 @@
  */
 int mo_version(struct Client* cptr, struct Client* sptr, int parc, char* parv[])
 {
-  struct Client *acptr;
   char *target=parv[1];
 
-  if (MyConnect(sptr) && parc > 1)
-  {
-    /* Send version request to all servers when * is given as parameter */
-    if (target[0] == '*' && target[1] == '\0') { 
-      sendcmdto_serv_butone(sptr, CMD_VERSION, cptr, ":%s", target);
-      send_reply(sptr, RPL_VERSION, version, debugmode, cli_name(&me),
-	         debug_serveropts());
-	    return 0;
-    }
-    if (!(acptr = find_match_server(target))) {
-      send_reply(sptr, ERR_NOSUCHSERVER, target);
-      return 0;
-    }
-    target = cli_name(acptr);
+  /* Send version request to all servers when * is given as parameter */
+  if (parc > 1 && target[0] == '*' && target[1] == '\0') { 
+    sendcmdto_serv_butone(sptr, CMD_VERSION, cptr, ":%s", target);
+    send_reply(sptr, RPL_VERSION, version, debugmode, cli_name(&me),
+      debug_serveropts());
+    return 0;
   }
 
   if (hunt_server_cmd(sptr, CMD_VERSION, cptr, feature_int(FEAT_HIS_REMOTE),
-                                                           ":%C", 1,
-                                                           parc, parv)
-                      == HUNTED_ISME)
+    ":%C", 1, parc, parv) == HUNTED_ISME)
   {
     send_reply(sptr, RPL_VERSION, version, debugmode, cli_name(&me),
 	       debug_serveropts());
@@ -166,21 +155,10 @@
  */
 int ms_version(struct Client* cptr, struct Client* sptr, int parc, char* parv[])
 {
-  struct Client *acptr;
   char *target=parv[1];
 
-  if (MyConnect(sptr) && parc > 1)
-  {
-    if (!(acptr = find_match_server(target)))
-    {
-      send_reply(sptr, ERR_NOSUCHSERVER, target);
-      return 0;
-    }
-    target = cli_name(acptr);
-  }
-
   /* Version request with parameter *, send my version info and pass on the request */
-  if (target[0] == '*' && target[1] == '\0') 
+  if (parc > 1 && target[0] == '*' && target[1] == '\0') 
   {
     sendcmdto_serv_butone(sptr, CMD_VERSION, cptr, ":%s", target);
     send_reply(sptr, RPL_VERSION, version, debugmode, cli_name(&me),
